src/System.Text.Json/src/System/Text/Json/JsonReaderState.cs

```diff
@@ -62,12 +51,11 @@ public struct JsonReaderState
         /// across async/await boundaries and hence this type is required to provide support for reading
         /// in more data asynchronously before continuing with a new instance of the <see cref="Utf8JsonReader"/>.
         /// </remarks>
-        public JsonReaderState(int maxDepth = StackFreeMaxDepth, JsonReaderOptions options = default)
+        public JsonReaderState(int maxDepth = CustomUncheckedBitArray.AllocationFreeMaxDepth, JsonReaderOptions options = default)
```

Comment:
The way the code is structured, it's not immediately obvious to someone changing the value of the internal AllocationFreeMaxDepth that they're actually changing a public value that's part of the API surface area/contract.

Reply:
Decoupled by adding a separate `DefaultMaxDepth` constant.